src/metainfo.rs

```diff
@@ -0,0 +1,222 @@
+use bencode::{Bencode, BDict, BList, BString, BInt};
+use std::{error, fmt};
+
+#[derive(Default)]
+pub struct MetaInfo {
+    pub announce: String,
+    pub announce_list: Vec<String>,
```

Comment:
I think this is supposed to be a list of lists of strings, as in: `Vec<Vec<String>>` as per https://wiki.theory.org/BitTorrentSpecification#Metainfo_File_Structure 
I also looked at this [Arch Linux torrent](https://www.archlinux.org/releng/releases/2016.08.01/torrent/) inside [QT Torrent Editor](https://sourceforge.net/projects/torrent-file-editor) ('Raw' tab) and I see nested lists:

    "announce-list": [
        [
            "http://tracker.archlinux.org:6969/announce"
        ]
    ],